### DISCUSSION TITLE
Constructor for Surenoo SLG12864I LCD Display

### DISCUSSION THREAD

**Discussion by johngineer** [olikraus/u8g2#1688] [Ideas]
I recently received a couple examples of this display: https://www.aliexpress.com/item/32913297404.html?spm=a2g0s.12269583.0.0.9f0e3b9fTgWb2X

The MPN is given as SLG12864I. It uses an ST7567 COG driver.

I didn't find it in the current list of constructors but I was able to get it working using this one:

`U8G2_ST7567_JLX12864_F_4W_HW_SPI u8g2(U8G2_MIRROR, PIN_LCD_CS, PIN_LCD_A0, PIN_LCD_RST);`

However, the display bitmap is still flipped about both the X and Y axes unless I call 

`u8g2.setFlipMode(1);`

Is it possible to use this information to create a dedicated constructor for the SLG display? I suspect the driver is mostly the same except for the scan direction or byte order.

**Comment by olikraus** [comment]
Created issue #1690
